### ISSUE TITLE
Add averaging function for quaternions

### ISSUE THREAD

**Issue by christophhagen** [KieranWynn/pyquaternion#11]
When working with noisy orientation data averaging is often needed to combine multiple orientations. Quaternions are difficult to use for this, since there is no trivial average. However, I found a paper that explains how to do an average and a weighted average:

Markley, F. Landis, Yang Cheng, John Lucas Crassidis, and Yaakov Oshman.
"Averaging quaternions." Journal of Guidance, Control, and Dynamics 30, no. 4 (2007): 1193-1197.
Link: https://ntrs.nasa.gov/archive/nasa/casi.ntrs.nasa.gov/20070017872.pdf

There is also a Matlab implementation:
Tolga Birdal. "averaging_quaternions" Matlab code.
http://jp.mathworks.com/matlabcentral/fileexchange/40098-tolgabirdal-averaging-quaternions

I wrote a two python functions that use numpy:
https://github.com/christophhagen/averaging-quaternions

It would be nice to have this feature integrated in this library.

**Comment by rpapallas** [comment]
@KieranWynn Do you think we could have this implemented in pyquaternion? If so, I can create a PR myself.

**Comment by KieranWynn** [comment]
Absolutely. If you put up a PR, I'd be happy to review it for inclusion. 

Can you:
1. Reference the original paper in the implementation by providing a link in the comments
2. Include tests for your implementation in [pyquaternion/test/test_quaternion.py](https://github.com/KieranWynn/pyquaternion/blob/master/pyquaternion/test/test_quaternion.py)
3. Add documentation to the appropriate section of [docs/index.md](https://github.com/KieranWynn/pyquaternion/blob/master/docs/index.md)

**Comment by nag92** [comment]
I am working on the project that also requires an averaging of quaterians, was this ever completed?
